src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
```

Comment:
Here you are breaking **DRY (Don’t Repeat Yourself )** principle - almost the same code is in **calculateWomenAverageAge**, and **calculateAverageAgeDiff** functions. 

You can make new function : **countAverageAge** and use it in both functions. 

[read about DRY principle ](https://thevaluable.dev/dry-principle-cost-benefit-example/)